src/battle_util.c

```diff
@@ -7058,6 +7060,22 @@ static u8 ItemEffectMoveEnd(u32 battlerId, u16 holdEffect)
             effect = ITEM_EFFECT_OTHER;
         }
         break;
+    case HOLD_EFFECT_BERSERK_GENE:
+        BufferStatChange(battlerId, STAT_ATK, STRINGID_STATROSE);
+        gEffectBattler = battlerId;
+        if (gBattleMons[gEffectBattler].ability != ABILITY_OWN_TEMPO
+         || IsBattlerTerrainAffected(battlerId, STATUS_FIELD_MISTY_TERRAIN))
```

Comment:
I think these should use `CanBeConfused`.

Reply:
I disagree. If a confused mon passes its confusion with baton pass, berserk gene still activates and confuses infinitely. (technically for the remaining turns but we handle those differently). `CanBeConfused` checks for confusion and wouldnt allow this behavior